actions/actions.go

```diff
@@ -132,16 +132,19 @@ func (s *service) doWork(ctx context.Context) error {
 		iteration int
 	)
 
-	b := backoff.WithContext(backoff.WithMaxRetries(backoff.NewConstantBackOff(5*time.Second), 3), ctx)
-	errR := backoff.Retry(func() error {
+	boff := waitext.NewConstantBackoff(5 * time.Second)
+
+	errR := waitext.RetryWithContext(ctx, boff, 3, func(ctx context.Context) (bool, error) {
 		iteration++
 		actions, err = s.castAIClient.GetActions(ctx, s.k8sVersion)
 		if err != nil {
-			s.log.Errorf("polling actions: get action request failed: iteration: %v %v", iteration, err)
-			return err
+			return true, err
 		}
-		return nil
-	}, b)
+		return false, nil
+	}, func(err error) {
```

Comment:
Could we make the logging of the error the default? Then we don't need to pass 2nd "logNotify" function. If someone wants a custom log message they could just wrap returning error with as much details as they need using `fmt.Errorf`. What do you think?